components/org.wso2.carbon.identity.password.policy/src/main/java/org/wso2/carbon/identity/password/policy/handler/PasswordPolicyValidationHandler.java

```diff
@@ -165,6 +169,10 @@ public void handleEvent(Event event) throws IdentityEventException {
         try {
             policyRegistry.enforcePasswordPolicies(credentials.toString(), userName);
         } catch (PolicyViolationException e) {
+            if (PasswordPolicyStatusCodes.ERROR_CODE_PASSWORD_POLICY_VIOLATION
+                    .equals(e.getErrorCode())) {
+                throw IdentityException.error(IdentityEventException.class, e.getErrorCode(), e.getMessage(), e);
```

Comment:
If we are using the same error message passed from the generated exception, lets verify that it includes a message with necessary details. 
Since the purpose is to change the error, better to add a comment in the PR mentioning what the error message would look like.

Reply:
Error message varies depending on password policies defined here.